gui/src/vmm/debug/mod.rs

```diff
@@ -1,79 +1,96 @@
 // SPDX-License-Identifier: MIT OR Apache-2.0
-use super::cpu::CpuManager;
+use super::cpu::{CpuManager, GdbError};
 use crate::debug::DebugClient;
-use crate::error::RustError;
 use crate::hv::Hypervisor;
 use crate::screen::Screen;
 use gdbstub::stub::state_machine::state::{Idle, Running};
 use gdbstub::stub::state_machine::{GdbStubStateMachine, GdbStubStateMachineInner};
 use gdbstub::stub::MultiThreadStopReason;
+use thiserror::Error;
 
-pub fn dispatch_idle<H: Hypervisor, S: Screen>(
-    target: &mut CpuManager<H, S>,
-    mut state: GdbStubStateMachineInner<
-        'static,
-        Idle<CpuManager<H, S>>,
-        CpuManager<H, S>,
-        DebugClient,
-    >,
-) -> Result<
-    Result<
-        GdbStubStateMachine<'static, CpuManager<H, S>, DebugClient>,
-        GdbStubStateMachineInner<'static, Idle<CpuManager<H, S>>, CpuManager<H, S>, DebugClient>,
-    >,
-    RustError,
-> {
-    // Check for pending command.
-    let b = match state.borrow_conn().read() {
-        Ok(v) => v,
-        Err(e) if e.kind() == std::io::ErrorKind::WouldBlock => return Ok(Err(state)),
-        Err(e) => {
-            return Err(RustError::with_source(
-                "couldn't read data from the debugger",
-                e,
-            ));
+impl<H: Hypervisor, S: Screen> CpuManager<H, S> {
+    pub fn dispatch_idle(
```

Comment:
This should be private. Name also does not make sense since it become a method on `CpuManager`. Probably rename to `dispatch_gdb_idle`.